test/e2e/e2e_test.go

```diff
@@ -200,6 +202,90 @@ func TestE2EFlow(t *testing.T) {
 		VerifyIncreaseOfUserAccountCount(t, originalToolchainStatus, currentToolchainStatus, targetedJohnMur.Spec.UserAccounts[0].TargetCluster, 1)
 	})
 
+	t.Run("verify userAccount is not deleted if namespace is not deleted", func(t *testing.T) {
+		// given
+		johnMUR, err := hostAwait.WaitForMasterUserRecord(johnsmithName)
+		require.NoError(t, err)
+
+		fmt.Printf("\n >>>>>>>>>>>>>>>>>>>>>>>>>>>> MUR fetched>>>>>>>>>>>>>>>>")
+		// set deletion timestamp and add finalizer
+		deletionTS := metav1.Now()
+		johnMUR.DeletionTimestamp = &deletionTS
+		coputil.AddFinalizer(johnMUR, "finalizer.toolchain.dev.openshift.com")
+		if err := hostAwait.Client.Update(context.TODO(), johnMUR); err != nil {
+			fmt.Errorf("failed to set deletionTS and finalizer")
```

Comment:
here you should do something like `require.NoError(t, err)` instead of just printing the error, because otherwise the code will continue to execute.